vaultManager/vaultManager.go

```diff
@@ -63,34 +99,58 @@ func baseVaultPath(dataType string) string {
 	return fmt.Sprintf("%s/%s/%s", remote_signer.VaultBackend, dataType, remote_signer.VaultNamespace)
 }
 
-func (vm *VaultManager) getClient() *api.Client {
+func (vm *VaultManager) validTokenTTL() bool {
+	if vm.token.getTime != nil {
+		var now = time.Now().Unix()
+		var timeWithTTL = vm.token.getTime.Unix() + int64(vm.token.ttl.Seconds())
```

Comment:
The value of `timeWithTTL`  my cause the usage of an expired token, as it's pointing to the exact time the token expire. A common practice for avoiding this issue is to calculate this value with some minutes before the actual expiration:
```suggestion
		var timeWithTTL = vm.token.getTime.Add(vm.token.ttl).Add(time.Minute*-1)
```